pkg/k8s/synced/crd.go

```diff
@@ -206,9 +211,27 @@ func SyncCRDs(ctx context.Context, clientset client.Clientset, crdNames []string
 
 func (s *crdState) add(obj interface{}) {
 	if pom := k8s.CastInformerEvent[slim_metav1.PartialObjectMetadata](obj); pom != nil {
-		s.Lock()
-		s.m[CRDResourceName(pom.GetName())] = true
-		s.Unlock()
+		vsnstr := pom.GetLabels()[k8sconst.CustomResourceDefinitionSchemaVersionKey]
+		vsn, err := semver.ParseTolerant(vsnstr)
+		if err != nil {
+			log.WithError(err).WithFields(logrus.Fields{
+				logfields.CRD:     pom.GetName(),
+				logfields.Version: vsnstr,
+			}).Warning("Failed to parse CRD version")
+			return
+		}
+
+		if vsn.GE(semver.MustParse(k8sconst.CustomResourceDefinitionSchemaVersion)) {
```

Comment:
Nit: maybe better to compute `semver.MustParse(k8sconst.CustomResourceDefinitionSchemaVersion)` once at import time and save it in a package level variable?